blueocean-rest-impl/src/test/java/io/jenkins/blueocean/service/embedded/rest/UserImplPermissionTest.java

```diff
@@ -111,6 +115,21 @@ public boolean hasPermission(Authentication a, Permission permission) {
         when(Jenkins.getAuthentication()).thenReturn(authentication);
         when(Jenkins.getInstance()).thenReturn(jenkins);
 
+        try {
+            // After Jenkins 2.77 hasPermission is no longer in Node.class and is not final so we need to mock it
+            // prior to it is called as being final and mocking it will fail for the same reason.
+            // TODO remove after core base line is >= 2.77
+            Node.class.getDeclaredMethod("hasPermission", Permission.class);
+        } catch (NoSuchMethodException e) {
```

Comment:
Not proud of this but if we want it to work with the current core and the PCT we need to check it.

Reply:
We've all done worse things than this ;) it is test code anyway - and lovingly documented, wonderful catch @alvarolobato !